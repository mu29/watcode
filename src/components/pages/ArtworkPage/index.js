import React from 'react'
import PropTypes from 'prop-types'
import { Header, GenericTemplate } from 'components'
import { ArtworkView } from 'containers'

const ArtworkPage = ({ match }) => (
  <GenericTemplate header={ <Header location={ `/artworks/${match.params.id}` } /> }>
    <ArtworkView id={ match.params.id } />
  </GenericTemplate>
)

ArtworkPage.propTypes = {
  match: PropTypes.shape({}).isRequired,
}

export default ArtworkPage
