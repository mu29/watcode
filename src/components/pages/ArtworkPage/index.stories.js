import React from 'react'
import { storiesOf } from '@storybook/react'
import ArtworkPage from '.'

storiesOf('Pages/ArtworkPage', module)
  .add('기본', () => (
    <ArtworkPage />
  ))
