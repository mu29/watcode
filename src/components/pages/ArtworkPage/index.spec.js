import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'
import ArtworkPage from '.'

const wrap = (props = {}) => shallow(<ArtworkPage match={{ params: { id: 1 } }} { ...props } />)

describe('<ArtworkPage />', () => {
  it('정상적으로 렌더링된다', () => {
    const wrapper = wrap()
    expect(wrapper).toMatchSnapshot()
  })
})
