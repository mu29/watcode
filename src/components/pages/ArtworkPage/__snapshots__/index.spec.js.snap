// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArtworkPage /> 정상적으로 렌더링된다 1`] = `
<GenericTemplate
  header={
    <Header
      location="/artworks/1"
    />
  }
>
  <ArtworkView
    id={1}
  />
</GenericTemplate>
`;
